### ISSUE TITLE
no-fallthrough: intentional fall through comment placement

### ISSUE THREAD

**Issue by larsgw** [eslint/eslint#10608]
**What rule do you want to change?**
`no-fallthrough`

**Does this change cause the rule to produce more or fewer warnings?**
Fewer

**How will the change be implemented? (New option, new default behavior, etc.)?**
New default behavior: the `// fall through` comment doesn't have to be the last comment before the next `case`, but it has to be after the last line of actual code.

**Please provide some example code that this change will affect:**

```js
switch (dateParts.length) {
  // Day
  case 3:
    dateParts[2] = dateParts[2].padStart(2, '0')
    // fall through

  // Month
  case 2:
    dateParts[1] = dateParts[1].padStart(2, '0')
    // fall through

  // Year
  case 1:
    dateParts[0] = dateParts[0].padStart(4, '0')
    break
}
```

**What does the rule currently do for this code?**

Error on the first two cases (`Expected a 'break' statement before 'case'.`), because the comments labelling the next `case` (`// Month` and `// Year`) make it so that the respective `// fall through` comments don't count.

**What will the rule do after it's changed?**

Not error.

**Comment by larsgw** [comment]
I'll try to get started on a PR, then.
